File: OPENAPI-REST-API/openapi-client/javascript-closure-angular/API/Client/ServerAssets.js
goog.provide('API.Client.ServerAssets');

/**
 * @record
 */
API.Client.ServerAssets = function() {}

/**
 * The title of the assets.
 * @type {!string}
 * @export
 */
API.Client.ServerAssets.prototype.title;

/**
 * The size of the assets.
 * @type {!number}
 * @export
 */
API.Client.ServerAssets.prototype.size;

/**
 * The type of the assets.
 * @type {!string}
 * @export
 */
API.Client.ServerAssets.prototype.type;

/**
 * The header of the assets table.
 * @type {!Array<!string>}
 * @export
 */
API.Client.ServerAssets.prototype.header;

/**
 * The rows of the assets table.
 * @type {!Array<!Array<!string>>}
 * @export
 */
API.Client.ServerAssets.prototype.rows;

File: OPENAPI-REST-API/openapi-client/scalaz/src/main/scala/org/openapitools/client/api/ServerAssets.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ServerAssets._

case class ServerAssets (
  /* The title of the assets. */
  title: Option[String],
/* The size of the assets. */
  size: Option[Integer],
/* The type of the assets. */
  `type`: Option[String],
/* The header of the assets table. */
  header: Option[List[String]],
/* The rows of the assets table. */
  rows: Option[List[List[String]]])

object ServerAssets {
  import DateTimeCodecs._

  implicit val ServerAssetsCodecJson: CodecJson[ServerAssets] = CodecJson.derive[ServerAssets]
  implicit val ServerAssetsDecoder: EntityDecoder[ServerAssets] = jsonOf[ServerAssets]
  implicit val ServerAssetsEncoder: EntityEncoder[ServerAssets] = jsonEncoderOf[ServerAssets]
}
